tests/utils/profilehandler/data_preparation.go

```diff
@@ -0,0 +1,222 @@
+package profilehandler
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/tests/utils/common"
+	CON "github.com/openshift/rosa/tests/utils/common/constants"
+	"github.com/openshift/rosa/tests/utils/exec/rosacli"
+)
+
+func PrepareVersion(client *rosacli.Client, versionRequirement string, channelGroup string, hcp bool) (
+	*rosacli.OpenShiftVersionTableOutput, error) {
+	versionList, err := client.Version.ListAndReflectVersions(channelGroup, hcp)
+	if err != nil {
+		return nil, err
+	}
+
+	if CON.VersionLatestPattern.MatchString(versionRequirement) {
+		return versionList.Latest()
+	} else if CON.VersionMajorMinorPattern.MatchString(versionRequirement) {
+		version, err := versionList.FindNearestBackwardMinorVersion(versionRequirement, 0, true)
+		return version, err
+	} else if CON.VersionRawPattern.MatchString(versionRequirement) {
+		return &rosacli.OpenShiftVersionTableOutput{
+			Version: versionRequirement,
+		}, nil
+	} else if CON.VersionFlexyPattern.MatchString(versionRequirement) {
+		return nil, nil // Implement later
```

Comment:
Do we have any ticket to track this implementation? If so would be good to link in the comment. Also a NP, would be good prefix the comment with `TODO` so it stands out and easy to grep across the code base.

Reply:
Updated to TODO. No tickets tracking, the day1 is keep ongoing. Just merge this one so tear down can start. It will be implemented in coming PR.